jenkins/run

```diff
@@ -16,12 +33,33 @@ if [[ -z "${IMAGE_REPO}" ]]; then
     IMAGE_REPO="quay.io/pbench"
 fi
 
+IHOME=/container/home
+
+if [[ -z "${CONTAINER_HOME}" ]]; then
+    HOME_VOLUME=${IHOME}
+else
+    HOME_VOLUME="${CONTAINER_HOME}:${IHOME}:${CONTAINER_HOME_OPTION:="z"}"
```

Comment:
I would use `:-` rather than `:=` for `CONTAINER_HOME_OPTION`, but I guess either will work, here.

Reply:
Well, shoot, I didn't think of that.

Technically someone could have `export CONTAINER_HOME_OPTION=` and now it'd come back as "z" instead of just affecting the expansion within `run`. Not a particularly realistic scenario, but I guess you're right.

Reply:
> someone could have `export CONTAINER_HOME_OPTION=` and now it'd come back as "z" instead of just affecting the expansion within run

Assuming that `jenkins/run` is invoked as a command, it will be run in a subshell, with its own environment.  If line 41 is executed, then after that point, `CONTAINER_HOME_OPTION` will be defined (regardless of whether it was before or not), and it will either still have its original value or it will have the value `"z"` if it was previously null or unset.  But, as we're in a subprocess where we control the code, probably no one will notice or care.

Now, inside the container, which we've set to inherit the parent process environment, `CONTAINER_HOME_OPTION` will be set (to `"z"`, if nothing else) any time `CONTAINER_HOME` is defined.  I don't know if that's useful or not, but there it is.  Switching from `:=` to `:-` would mean that `CONTAINER_HOME_OPTION` would only be defined inside the container when it was explicitly defined outside the container...I don't know if that's better or not.

BTW, I don't think that setting `CONTAINER_HOME_OPTION` to an empty value makes the `:=` expansion behave differently from the way it does when the variable is unset -- either way I think the variable ends up set to `"z"`...and perhaps that's a "feature".

Reply:
Yes, you're correct that the value won't be propagated back out of the script. I got carried away, there. 😆 

For the last paragraph; right, but my point was to contrast `-` and `:-` with respect to a variable that's *defined* but blank. (That is, `:-` will set `CONTAINER_HOME_OPTION=''` to `z` while `-` won't. Similarly for `:=` and `=`.)

Reply:
I never noticed before the sentence at the end of the paragraph in the man page which says, "omitting the colon results in a test only for a parameter that is unset."  That's good to know....

> That is, `:-` will set `CONTAINER_HOME_OPTION=''` to `z` while `-` won't.

I agree iff you change "set" to "evaluate" (but, I concur with your assertion about `:=` and `=`).